lighthouse/classes/plate_event.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC, abstractmethod
+from datetime import datetime
+from enum import Enum, auto
+from typing import Any, Dict, List, Optional, Union
+from uuid import uuid4
+
+from flask import current_app as app
+
+from lighthouse.constants.fields import (
+    FIELD_LAB_ID,
+    FIELD_LH_SAMPLE_UUID,
+    FIELD_RESULT,
+    FIELD_RNA_ID,
+    FIELD_ROOT_SAMPLE_ID,
+)
+from lighthouse.messages.message import Message
+from lighthouse.types import EventMessage, SampleDoc, Subject
+
+
+class PlateEvent(ABC):
+    ###
+    # subjects and roles for the message to the events warehouse
+    ###
+    ROLE_TYPE_ROBOT = "robot"
+    ROLE_TYPE_SAMPLE = "sample"
+    ROLE_TYPE_CP_SOURCE_LABWARE = "cherrypicking_source_labware"
+    SUBJECT_TYPE_SAMPLE = ROLE_TYPE_SAMPLE
+    SUBJECT_TYPE_ROBOT = ROLE_TYPE_ROBOT
+    SUBJECT_TYPE_PLATE = "plate"
+
+    class PlateTypeEnum(Enum):
+        SOURCE = auto()
+        DESTINATION = auto()
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def plate_barcode(self) -> str:
+        return self._plate_barcode
+
+    def __init__(self, name: str, plate_type: PlateTypeEnum) -> None:
+        self._name = name
+        self._plate_type = plate_type
+        self._plate_barcode = ""
+
+    @abstractmethod
+    def initialize_event(self, params: Dict[str, Union[str, Any]]) -> None:
+        ...
+
+    @abstractmethod
+    def _create_message(self) -> Message:
+        ...
```

Comment:
If the method is private and abstract, but is not called anywhere in this class, then why is important to define it when inheriting? What happens if we define it but we don't use it on inheriting? 
I think we are missing here an implementation of a public method using this private inside this class.

Reply:
Same comment for _send_warehouse_message

Reply:
Yes, good points! Will add some implementation to `process_event()` which uses the abstract methods.